testing/lib/mock_from_registry.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from containerregistry.client.v2_2 import docker_image
+import mock_registry
+
+
+class MockFromRegistry(docker_image.FromRegistry):
+
+    REGISTRY = mock_registry.MockRegistry()
+
+    def __init__(self, name="", basic_creds=None, transport=None):
+        self._name = name
+
+    def setRegistry(self, registry):
+        if isinstance(registry, mock_registry.MockRegistry):
+            MockFromRegistry.REGISTRY = registry
```

Comment:
I think this is setting this as a class variable, not on the instance itself. Should you set this as self. registry instead of MockFromRegistry.REGISTRY?

Reply:
I set it as a class variable so that if an image was pushed from session.Push then FromRegistry could still access it. I wasn't sure if there was a better way to do this?

Reply:
Ah I see - I think in that case it should probably be up to the "WithMockRegistry" class to create a MockRegistry instance and store it.

Does that make sense?

Reply:
I think so, is that the same as what I was doing in the tests [here](https://github.com/priyawadhwa/runtimes-common/blob/c84ddfdc69d725a9dcb7bdac4cacdc196c430a84/testing/lib/mock_registry_tests.py#L30)? I can move it into "WithMockRegistry" if so.

Reply:
Yup!

Reply:
I ended up moving it into a separate file for clarity!